src/lib/utils.ts

```diff
@@ -174,3 +178,34 @@ export const convertUnixTimeToFromNowFormat = (unixTime: number): string =>
 export const removeLeadingZeros = (value: string) => {
   return value.replace(/^0+/, '')
 }
+
+/**
+ * Creates a pending tx object from a wallet connect response
+ * @param wcResponse wallet connect response
+ * @param params additional required params to create a pending tx
+ * @returns pending tx object
+ */
+export const createPendingTxFromWcResponse = async (
+  wcResponse: TransactionResponse,
+  { chainId, from, to }: { chainId: number; from: string; to: string },
+) => {
+  try {
+    const abiEnhancer = new AbiEnhancer()
+    const enhancedTx = await abiEnhancer.enhance(chainId, {
+      data: wcResponse.data,
+    })
+    return {
+      ...(wcResponse as any),
```

Comment:
I think it's forbidden to use the any type, what you can do is ...response then the missing properties you can add them as '', such as transactionIndex, txType, etc...